lib/Mojo/IRC.pm

```diff
@@ -492,7 +492,7 @@ sub irc_nick {
   my ($self, $message) = @_;
   my $old_nick = ($message->{prefix} =~ /^[~&@%+]?(.*?)!/)[0] || '';
 
-  if ($old_nick eq $self->nick) {
+  if (lc $old_nick eq lc $self->nick) {
```

Comment:
Also, this relaxes the nick change handler because nicks are not case sensitive.